### ISSUE TITLE
Re-enable rules that are covered by eslint-plugin-babel

### ISSUE THREAD

**Issue by gajus** [gajus/eslint-config-canonical#1]
Some of the ESLint rules when used against ES6 code produce false-positives. These cases are fixed using https://github.com/babel/eslint-plugin-babel. The specific rules that are affected are:

```
"generator-star-spacing": 1,
"new-cap": 1,
"object-curly-spacing": 1,
"object-shorthand": 1,
"arrow-parens": 1
```

These rules are replaced with:

```
"babel/generator-star-spacing": 1,
"babel/new-cap": 1,
 "babel/object-curly-spacing": 1,
"babel/object-shorthand": 1,
"babel/arrow-parens": 1
```

This task is a reminder to remove `eslint-plugin-babel` when the inbuilt ESLint rules work against ES6 codebase.
